tests/cases/fourslash/jsDocFunctionSignatures2.ts

```diff
@@ -8,4 +8,4 @@
 ////
 //// f6('', /**/false)
 
-verify.signatureHelp({ marker: "", text: "f6(arg0: string, arg1?: boolean): number" });
+verify.signatureHelp({ marker: "", text: "f6(arg0: string, arg1?: boolean | undefined): number" });

```

Comment:
Is this extra `| undefined` a behavior change for jsdoc types? I'm not super familiar with the semantics of `=` in this parsing to know if this is a good or bad change.

Reply:
Yes. A jsdoc `type=` always has `| undefined` mixed into its' type (this is how it's typechecked, regardless of where in the parameter list it lies, though it is also *usually* considered an optional function parameter). Since we reuse the input node here, we transform the `type=` into `type | undefined` to strip the jsdoc syntax away.

Reply:
Of note: For parameters, `p?: boolean` and `p?: boolean | undefined` are functionally identical (since `exactOptionalProperties` only affects properties). Hence why the generated output based on the type omitted the `| undefined` before (it's redundant), while now that we're cloning/transforming the input node, we're being explicit about the `| undefined`. (Since a jsdoc `=` can appear outside of a parameter type position, unlike a parameter `?`, it really is just identical to adding a `| undefined` but with the extra baggage of also making a parameter that uses it optional)

Reply:
man, jsdoc is weird